core/src/test/java/org/hibernate/ogm/backendtck/embeddable/EmbeddableTest.java

```diff
@@ -292,6 +292,56 @@ public void testPersistWithListEmbeddedInNestedComponent() throws Exception {
 		session.close();
 	}
 
+	@Test
+	public void testNestedEmbeddedWithNullProperties() {
+		final Session session = openSession();
+
+		// persist entity without the embeddables
+		Transaction transaction = session.beginTransaction();
+		Account account = new Account();
+		account.setLogin( "gunnar" );
+		account.setHomeAddress( new Address() );
+		account.getHomeAddress().setCity( "Lima" );
+		account.getHomeAddress().setType( new AddressType( "primary" ) );
+		session.persist( account );
+
+		transaction.commit();
+		session.clear();
+
+
+		// set nested embedded to null
+		transaction = session.beginTransaction();
+		Account loadedAccount = (Account) session.get( Account.class, account.getLogin() );
+		loadedAccount.getHomeAddress().setType( null );
+		transaction.commit();
+		session.clear();
+
+		// read back nested embedded and set regular embedded to null
+		transaction = session.beginTransaction();
+		loadedAccount = (Account) session.get( Account.class, account.getLogin() );
+		assertThat( loadedAccount ).as( "Cannot load persisted object with nested embeddedables" ).isNotNull();
+		assertThat( loadedAccount.getHomeAddress() ).isNotNull();
```

Comment:
Should we have a test in _mongodb_ which asserts the actual persistent structure at this point? I think that'd make sense. There are some *Mapping tests in _mongodb_.